spec/enumerables_spec.rb

```diff
@@ -0,0 +1,148 @@
+require './lib/enumerables'
+
+RSpec.describe Enumerable do
+    let(:arr) { [1, 2, 3, 4, 3] }
+    let(:new_arr) { [ ] }
+    let(:arg) { 23 }
+    let(:count) { 5 }
+    let(:word) {['ChristLord', 'Living', 'Life', 'Love']}
+
+
+    describe '#.my_each' do
+
+        context 'When array and block are given'do
+            it 'returns the items in the array' do
+                expect( arr.my_each_with_index { |i, v| i}).to eql( arr.each_with_index { |i, v| i} )
+            end
+        end
+
+        context 'When no block is given' do
+            it 'returns an enumerator' do
+                expect(arr.my_each).to be_instance_of(Enumerator)
+            end
+        end
+    end
+
+    describe '#.my_each_with_index' do
+        context 'When no block is given' do
+            it 'returns an enumerator' do
+                expect(arr.my_each_with_index).to be_instance_of(Enumerator)
+            end
+        end
+    end
+
+    describe '#.my_select' do
+
+        context 'When array and block are given'do
+            it 'returns the selected items in the array' do
+                 expect( arr.my_select { |i| i > 3 }).to eql(arr.select { |i| i > 3})
+            end
+        end
+        context 'When no block is given' do
+            it 'returns an enumerator' do
+                 expect( arr.my_select {|i| i == 3 }).to eql( [ 3, 3 ] )
+                 expect(arr.my_select {|i| i == 3 }).to eql(arr.my_select {|i| i == 3 })
+            end
+        end
+     end
+
+    describe '#.my_all?' do
+
+        context 'When array and block are given'do
+
+            it 'returns false when not all the items in the array are true' do
+                expect( arr.my_all? { |i| i.even? }).to eql(false)
+            end
```

Comment:
Well, the actual task of the test here is to check none of the items are nil/false. The below format is enough to check it I think.
```ruby
it 'returns false when not all the items in the array are true' do
  expect( arr.my_all?).to eql(arr.all?)
end
```